### PR TITLE
Add dependent auth to lab

### PR DESCRIPTION
**Pull request by leobcgov**
# Fixes or Implements [AB#9381](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9381)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Description
- Added dependent auth to lab

## Testing

* [X] Unit Tests Updated
* [ ] Functional Tests Updated
* [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

NO

### Browsers Tested

* [ ] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.  

</br>
</br>
</br>
___
  
## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"?

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

### CODE REVIEW

File: Apps/Database/src/Delegates/DBUserDelegateDelegate.cs [hunk 1]
```diff
@@ -92,5 +92,19 @@ public DBResult<UserDelegate> Delete(string ownerId, string delegateId, bool com
         {
             throw new System.NotImplementedException();
         }
+
+        /// <inheritdoc />
+        public bool Find(string ownerId, string delegateId)
```
Comment (tiagograf-AOT) [inline Apps/Database/src/Delegates/DBUserDelegateDelegate.cs:97]: i think "exists" make more sense as a name!

**Review by tiagograf-AOT — DISMISSED** [review]

File: Apps/Database/src/Delegates/DBUserDelegateDelegate.cs (hunk 1 above)
Comment (sslaws) [reply Apps/Database/src/Delegates/DBUserDelegateDelegate.cs:97]: Agreed, I said Find but we're not returning the object so exists makes more sense.

Comment (leobcgov) [reply Apps/Database/src/Delegates/DBUserDelegateDelegate.cs]: Updated the name to Exists

**Review by tiagograf-AOT — APPROVED** [review]

File: Apps/Common/src/AccessManagement/Authorization/Handlers/FhirResourceAuthorizationHandler.cs [hunk 2]
```diff
@@ -179,18 +183,28 @@ private bool IsDelegated(AuthorizationHandlerContext context, string resourceHDI
...
                     if (context.User.HasClaim(c => c.Type == GatewayClaims.HDID) && scopes.Intersect(userDelegatedScopes).Any())
                     {
-                        this.logger.LogError("user delgation is not implemented, returning not authorized");
-
-                        // TODO:  Future check needed for patient to patient delegation.
-                    }
-                    else
-                    {
-                        this.logger.LogWarning($"Patient delgation validation for Patient resource {resourceHDID} failed");
+                        string userHDID = context.User.FindFirst(c => c.Type == GatewayClaims.HDID).Value;
```
Comment (sslaws) [inline Apps/Common/src/AccessManagement/Authorization/Handlers/FhirResourceAuthorizationHandler.cs:186]: I think you should be in the block above right?

**Review by sslaws — CHANGES_REQUESTED** [review]

**Review by sslaws — APPROVED** [review]

**Pull request merged**